### PR TITLE
add getProfiledGraph API [Experimental] [WIP]

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: torch/csrc/jit/graph_executor.h [hunk 1]
```diff
@@ -49,6 +49,8 @@ struct TORCH_API GraphExecutor {
   std::shared_ptr<Graph> graph() const;
   GraphExecutorState getDebugState();
 
+  std::shared_ptr<Graph> getProfiledGraph() const;
```
Comment (houseroad) [inline torch/csrc/jit/graph_executor.h]: @spandantiwari  Nick has created some experimental API to ease your work.

This PR is still experimental, and probably won't be merged as it is.

Comment (spandantiwari) [reply torch/csrc/jit/graph_executor.h]: Thanks Nick and Lu.

Comment (Krovatkin) [reply torch/csrc/jit/graph_executor.h]: @spandantiwari  if you have more questions, please let me know! we could always chat more on slack

Comment (spandantiwari) [reply torch/csrc/jit/graph_executor.h]: @Krovatkin - thanks a lot. This is quite helpful. We just started looking at it in earnest. We will let you know if there's anything. Thanks for your support.

File: torch/csrc/jit/profiling_graph_executor_impl.cpp [hunk 2]
```diff
@@ -149,6 +149,13 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(Stack& stack) {
   return *optimized_plan_;
 }
 
+std::shared_ptr<Graph> ProfilingGraphExecutorImpl::_getProfiledGraph() const {
+  if (!pr_ || !optimized_plan_) {
```
Comment (BowenBao) [inline torch/csrc/jit/profiling_graph_executor_impl.cpp]: Hi @Krovatkin , thanks for providing this feature! I have been trying out this branch to get the profiled graph. But so far I'm only getting nullptr from this api. The `optimized_plan_` is always nullptr. I'm calling the api with the below snippet, there must be something I'm missing.
```c++
  m.def(
      "_jit_get_profiled_graph",
      [](std::shared_ptr<Graph> graph, const std::vector<at::Tensor>& inputs) {
        auto executor = GraphExecutor(graph);
        Stack stack(fmap<IValue>(inputs));
        executor.getPlanFor(stack);
        return executor.getProfiledGraph();
      });
```

Comment (Krovatkin) [reply torch/csrc/jit/profiling_graph_executor_impl.cpp]: @BowenBao could try running it twice  i.e. `executor.run(stack); executor.run(stack);` and then do `executor.getPlanFor()`. We definitely need to run it at least once, so we collect shapes information. Now that I'm looking at this API, I don't think we necessarily need to give you the optimized graph (i.e. making you run the executor twice). I'll push a patch so you can get the profiled graph after you run the executor once.

Comment (Krovatkin) [reply torch/csrc/jit/profiling_graph_executor_impl.cpp]: @BowenBao arggh, you are right this is wrong! I didn't really think python API through. let me push a patch to fix this!

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 4e6694b4:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=4e6694b405b72e8992f35f5bf9d24aa594cbb493).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4192602) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=82876092) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=82876092))
<details>
<summary>
<code>Automatic merge failed; fix conflicts and then commit the result.</code>
</summary>

```
CONFLICT (add/add): Merge conflict in .circleci/docker/common/install_android.sh 
Auto-merging .circleci/docker/common/install_android.sh 
CONFLICT (add/add): Merge conflict in .circleci/docker/build.sh 
Auto-merging .circleci/docker/build.sh 
CONFLICT (add/add): Merge conflict in .circleci/config.yml 
Auto-merging .circleci/config.yml 
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/pytorch_build_definitions.py 
Auto-merging .circleci/cimodel/data/pytorch_build_definitions.py 
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/caffe2_build_definitions.py 
Auto-merging .circleci/cimodel/data/caffe2_build_definitions.py 
Automatic merge failed; fix conflicts and then commit the result. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Pull request closed without merging**